lib/KeySharedPolicy.cc

```diff
@@ -51,7 +51,16 @@ KeySharedPolicy &KeySharedPolicy::setAllowOutOfOrderDelivery(bool allowOutOfOrde
 bool KeySharedPolicy::isAllowOutOfOrderDelivery() const { return impl_->allowOutOfOrderDelivery; }
 
 KeySharedPolicy &KeySharedPolicy::setStickyRanges(std::initializer_list<StickyRange> ranges) {
-    if (ranges.size() == 0) {
+    StickyRanges v;
+    for(StickyRange i:ranges)
+    {
+        v.push_back(i);
+    }
+    return this->setStickyRanges(v);
```

Comment:
I think we can convert a `std::initializer_list<T>` to a `std::vector<T>` implicitly.

Reply:
Done.